benchmark/cuchembench/inference/megamolbart.py

```diff
@@ -41,51 +41,67 @@ def is_ready(self, timeout: int = 10) -> bool:
         return True
 
     def smiles_to_embedding(self,
-                            smiles: str,
-                            pad_length: int):
-
-        embedding, pad_mask = self.megamolbart.smiles2embedding(smiles,
-                                                                pad_length=pad_length)
-        dim = embedding.shape
-        embedding = embedding.flatten().tolist()
-        return EmbeddingList(embedding=embedding,
-                             dim=dim,
-                             pad_mask=pad_mask)
-
+                            smiles: list,
+                            pad_length: int = None):
+
+        embedding, pad_mask = self.megamolbart.smiles2embedding(smiles, pad_length=pad_length)
+        storage = []
+        for molecule in range(len(smiles)):
+            emb = embedding[:, molecule, :].unsqueeze(1)
+            dim = emb.shape
+            mask = pad_mask[:, molecule].unsqueeze(1)
+            elem = EmbeddingList(embedding=emb.flatten().tolist(), dim=dim, pad_mask=mask)
+            storage.append(elem)
+        return embedding, pad_mask, storage
+
+    #TODO: Method is unused
     def embedding_to_smiles(self,
                             embedding,
-                            dim: int,
-                            pad_mask):
+                            pad_mask,
+                            storage = None,
+                            batch_size = 100):
         '''
         Converts input embedding to SMILES.
         @param transform_spec: Input spec with embedding and mask.
         '''
         import torch
```

Comment:
Why not global import?
Also, if unused may delete or comment?

Reply:
Same logic as other inside import. This should be fixed in clean up branch